### PR TITLE
doc: local trees parameter documentation

### PR DESCRIPTION
**Pull request by ethanglaser**
## Description

Follow-up to https://github.com/uxlfoundation/scikit-learn-intelex/pull/2615 (and https://github.com/uxlfoundation/oneDAL/pull/3139). Adds documentation of additional parameter to SPMD forest estimators. Open to discussion on the best way to do this since I don't believe we have any prior references for this.

image

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

### CODE REVIEW

File: doc/sources/algorithms.rst [hunk 1]
```diff
@@ -473,6 +473,10 @@ Classification
        - ``criterion`` != `'gini'`
        - ``oob_score`` = `True`
        - ``sample_weight`` != `None`
+
+       **Additional parameters:**
+
+       - ``local_trees_mode`` (bool, default=False): Enables local trees mode for distributed training. `n_estimators` is per rank, with isolated learning occurring on each processor before merging into a single model. This mode is experimental but scales better than default. This parameter is specific to the SPMD implementation and is not present in the standard scikit-learn API.
```
Comment (ethanglaser) [inline doc/sources/algorithms.rst]:
```suggestion
       - ``local_trees_mode`` (bool, default=False): Enables local trees mode for distributed training. ``n_estimators`` is per rank, with isolated learning occurring on each processor before merging into a single model. This mode is experimental but scales better than default. This parameter is specific to the SPMD implementation and is not present in the standard scikit-learn API.
```

File: doc/sources/algorithms.rst [hunk 2]
```diff
@@ -525,6 +529,10 @@ Regression
        - ``criterion`` != `'mse'`
        - ``oob_score`` = `True`
        - ``sample_weight`` != `None`
+
+       **Additional parameters:**
+
+       - ``local_trees_mode`` (bool, default=False): Enables local trees mode for distributed training. `n_estimators` is per rank, with isolated learning occurring on each processor before merging into a single model. This mode is experimental but scales better than default. This parameter is specific to the SPMD implementation and is not present in the standard scikit-learn API.
```
Comment (ethanglaser) [inline doc/sources/algorithms.rst]:
```suggestion
       - ``local_trees_mode`` (bool, default=False): Enables local trees mode for distributed training. ``n_estimators`` is per rank, with isolated learning occurring on each processor before merging into a single model. This mode is experimental but scales better than default. This parameter is specific to the SPMD implementation and is not present in the standard scikit-learn API.
```

**Comment by david-cortes-intel** [comment]
@ethanglaser The only section that I'm aware of where extra parameters are documented is here:
https://uxlfoundation.github.io/scikit-learn-intelex/2025.7/guide/acceleration.html#random-forest

The title of the doc section doesn't match at all with the contents, but perhaps you could put it there for now next to the other extra parameters of decision trees, and then later we can revisit the structuring of the docs.

File: doc/sources/guide/acceleration.rst [hunk 3]
```diff
@@ -73,3 +73,10 @@ times, especially for larger data sets. However, due to the reduced fidelity of
 the data, the resulting model can present worse performance metrics compared to
 a model trained on the original data. In such cases, the number of bins can be
 increased with the ``max_bins`` parameter.
+
+Another parameter that can improve performance at large scale for Random Forest, 
+specifically the ``sklearnex.spmd.ensemble`` ``RandomForestClassifier`` and
```
Comment (david-cortes-intel) [inline doc/sources/guide/acceleration.rst:78]: Could use links to the sklearn docs of the classes here, as done elsewhere - e.g. ``` :obj:`sklearn.ensemble.RandomForestClassifier` ```

File: doc/sources/algorithms.rst [hunk 4]
```diff
@@ -487,6 +487,10 @@ Classification
        - ``criterion`` != `'gini'`
        - ``oob_score`` = `True`
        - ``sample_weight`` != `None`
+
+       **Additional parameters:**
+
+       - ``local_trees_mode`` (bool, default=False): Enables local trees mode for distributed training. ``n_estimators`` is per rank, with isolated learning occurring on each processor before merging into a single model. This mode is experimental but scales better than default. This parameter is specific to the SPMD implementation and is not present in the standard scikit-learn API.
```
Comment (david-cortes-intel) [inline doc/sources/algorithms.rst:493]: I'd say this is not very descriptive.

* Does it mean that the result has `n_estimators*n_ranks` trees?
* Does the data get moved across ranks, or does each rank use the data that it owns?
* Maybe could also refer to them as 'rank/nodes' as otherwise it might not be immediately clear what a 'rank' here refers to.

Comment (ethanglaser) [reply doc/sources/algorithms.rst:493]: Ideally we could point to oneDAL docs, where this functionality was implemented. @Alexandr-Solovev can we get this documented in oneDAL?

Comment (Alexandr-Solovev) [reply doc/sources/algorithms.rst:493]: I will create a JIRA task for update the docs, because for now we mention this parameter in onedal only in one place here:
https://github.com/uxlfoundation/oneDAL/blob/151df1d4b1e9c41b51997bc20a0544ada2bd51ec/cpp/oneapi/dal/algo/decision_forest/common.hpp#L519
But if it helps, I also can clarify David's questions:
Does it mean that the result has n_estimators*n_ranks trees? - No, oneDAL will split the n_trees across the ranks internally.
Does the data get moved across ranks, or does each rank use the data that it owns? - Each rank uses just local data(data that it owns).
